File: seeker/snippet/Task2.java
//date: 2023-02-28T17:08:15Z
//url: https://api.github.com/gists/eb58f202d7b94f2558b194c6a4ed60c1
//owner: https://api.github.com/users/IliyanOstrovski

public class Task2 {
    public static void main(String[] args) {
        String str1 = "abc";
        String str2 = "123";
        System.out.println(reverse(str1));
        System.out.println(reverse(str2));
    }

    public static String reverse(String s) {
        if (s == null || s.isEmpty()) {
            return s;
        } else {
            return reverse(s.substring(1)) + s.charAt(0);
        }
    }
}

File: seeker/snippet/task_2.py
#date: 2025-08-13T16:40:41Z
#url: https://api.github.com/gists/c16e305a437ee7787571bd8dbb39ee8c
#owner: https://api.github.com/users/DanielIvanov19

class Employee:
    def __init__(self, name, age):
        self.name = name
        self.age = age


def my_vars(obj):
    if hasattr(obj, "__dict__"):
        return obj.__dict__
    raise TypeError(f"{type(obj).__name__} няма __dict__")
